bionmr_utils/md/process/calc.py

```diff
@@ -0,0 +1,118 @@
+import numpy as np
+import pandas as pd
+from typing import Tuple, Dict, List, Callable
+from pyxmolpp2 import calc_autocorr_order_2
+from tqdm import tqdm
+
+
+def calc_autocorr(vectors: Dict[Tuple[str, str], np.ndarray],
+                  calc_autocorr_func: Callable[[np.ndarray], List[float]] = calc_autocorr_order_2,
+                  limit=-1
+                  ) -> Dict[Tuple[str, str], List[float]]:
+    """
+    Get auto-correlation from trajectory
+
+    :param vectors: dict of (rid, aname): VectorXYZ
+    :calc_autocorr_func: function calculates auto-correlation
+    :limit: length of auto-correlation function
+    :return dict of (rid, aname): auto-correlation
+    """
+    autocorr = {(rid, aname): calc_autocorr_func(vector, limit)
+                for (rid, aname), vector in vectors.items()
+                }
+    return autocorr
+
+
+def calc_inertia_tensor_vectors_autocorr(rotation_matrices: np.array,
+                                         path_to_vectors_csv: str,
+                                         limit: int = -1,
+                                         ) -> np.array:
+    """
+    Get auto-correlation for inertia vectors
+
+    :rotation matrices: rotation matrices as numpy array of shape (N, 3, 3), where N is length of trajectory
+    :path_to_vectors_csv: path to csv file with coulumn names (x, y, z, w),
+                          where (x, y, z) coordinatesand and weights (w) of vectors
+                          corresponding to the directions in space along which rotation is considered
+    :limit: length of auto-correlation function
+    :return auto-correlation function
+    """
+    nodes = pd.read_csv(path_to_vectors_csv)
+    xyz = nodes[["x", "y", "z"]].values
+    weights = nodes["w"]
+
+    vectors = xyz.dot(rotation_matrices)
+    numbers_of_vector = vectors.shape[0]
+
+    sum_acorr = None
+
+    for i, number_of_vector in enumerate(tqdm(range(numbers_of_vector), desc="calc autocorr")):
+        w = weights[i]
+        autocorr = np.array(calc_autocorr_order_2(vectors[number_of_vector, :, :], limit=limit)) * w
+        if sum_acorr is None:
+            sum_acorr = autocorr
+        else:
+            sum_acorr += autocorr
+    avg_acorr = sum_acorr / 4 / np.pi

```

Comment:
It's better to avoid simultaneous allocation of huge array. :warning: untested code follows:
```suggestion
    number_of_vectors = rotation_matrices.shape[0]

    sum_acorr = np.zeros(number_of_vectors)

    for r, w in zip(xyz, weights):
        vectors = r.reshape((3,1)).dot(rotation_matrices)
        autocorr = np.array(calc_autocorr_order_2(vectors, limit=limit))
        sum_acorr += autocorr * w
```